### PR TITLE
Feature/animations for enemy

### PR DESCRIPTION
**Pull request by calmchowder**
Implemented animations for enemy class

### CODE REVIEW

File: SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs [hunk 1]
```diff
@@ -40,18 +43,27 @@ public class MeleeEnemy : MonoBehaviour, IEnemy {
...
-        if (InRange()) {
-            if (canAttack()) {
-                Debug.Log("COLLIDED");
-                // Perform attack animation here and call Attack();
-                Attack();
+        // Check if enemy will attack or move
+        float curr_distance = CheckPlayerDistance();
+        if (curr_distance <= aggroDistance) {
+            if (curr_distance <= meleeDistance) {
+                if (canAttack()) {
```
Comment (axk4647) [inline SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs]: Put conditions into one if statement using & (better for branching, and style)

File: SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs [hunk 2]
```diff
@@ -64,37 +76,73 @@ public class MeleeEnemy : MonoBehaviour, IEnemy {
...
     public void Move() {
         ThisAgent.SetDestination(target.position);
+        SetWalkingAnimation();
     }
 
     /**
      * Play the death animation, then delete the model after it finishes
      */
     public void Death() {
         Animation death = GetComponent<Animation>(); // Need to get actual animations lol
```
Comment (axk4647) [inline SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs]: is this needed? you are setting the animation in the animator

File: SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs [hunk 3]
```diff
@@ -64,37 +76,73 @@ public class MeleeEnemy : MonoBehaviour, IEnemy {
...
+    }
+
+    /**
+     * Sets the corresponding walking animation based on angle to player
+     */
+    private void SetWalkingAnimation() {
+        Vector3 targetDir = target.position - transform.position;
+        float angle = Vector3.SignedAngle(targetDir, transform.forward, Vector3.up);
+
+        if (angle <= 43.75 && angle >= -43.75) {
```
Comment (ArturoKuang) [inline SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs]: store the number for angles in a variable (better for style, clarity, and easier to change)

**Review by ArturoKuang — COMMENTED** [review]
Looks good. Just little details i am nit picking about

File: SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs [hunk 4]
```diff
@@ -40,18 +43,27 @@ public class MeleeEnemy : MonoBehaviour, IEnemy {
             Death();
         }
 
-        if (InRange()) {
-            if (canAttack()) {
-                Debug.Log("COLLIDED");
-                // Perform attack animation here and call Attack();
-                Attack();
+        // Check if enemy will attack or move
+        float curr_distance = CheckPlayerDistance();
```
Comment (ArturoKuang) [inline SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs]: Put conditions into one if statement using && (better for branching, and style)

File: SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs (hunk 2 above)
Comment (ArturoKuang) [inline SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs]: Is this needed? you set the animation in the animator

**Review by ArturoKuang — APPROVED** [review]

**Pull request merged**